### PR TITLE
timer,ttl: add some metrics for timer framework and TTL timer syncing

### LINKED ISSUE #45610: Add metrics to timer framework

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #45610

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: timer/runtime/worker.go [hunk 1]
```diff
@@ -68,6 +70,13 @@ type hookWorker struct {
 	ch        chan *triggerEventRequest
 	logger    *zap.Logger
 	nowFunc   func() time.Time
+	// metrics for worker
+	triggerRequestCounter       prometheus.Counter
+	onPreSchedEventCounter      prometheus.Counter
```
Comment (hawkingrei) [inline timer/runtime/worker.go:75]: We can create metrics package to support metrics for timer. such as ```executor/metrics```.

Comment (lcwangchao) [reply timer/runtime/worker.go:75]: done

**Review by hawkingrei — APPROVED** [review]

**Review by YangKeao — APPROVED** [review]

**Pull request merged**